### PR TITLE
Optional yard

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: build-tools/aminclude/autodocs-ycp.ami [hunk 1]
```diff
@@ -47,4 +47,10 @@ index.html: $(AUTODOCS_YCP) $(AUTODOCS_PM) $(AUTODOCS_RB)
 		pod2html --title=$$b --infile=$$i --outfile=$$b.html; \
 		echo "<a href='$$b.html'>$${b/__/::}</a>" > $$b.mod.html; \
 	done
-	if [ -n "$(AUTODOCS_RB)" ]; then $(YARD) doc --title "@RPMNAME@ -- Development Documentation" $(YARD_OPTIONS) $(AUTODOCS_RB); fi
+	if [ -n "$(AUTODOCS_RB)" ]; then \
+          if which yard 2>/dev/null; then \
+            $(YARD) doc --title "@RPMNAME@ -- Development Documentation" $(YARD_OPTIONS) $(AUTODOCS_RB);\
+          else \
+            echo "yard no available, skipping"; \
```
Comment (mvidner) [inline build-tools/aminclude/autodocs-ycp.ami]: *not*

File: package/yast2-devtools.changes [hunk 2]
```diff
@@ -1,4 +1,12 @@
 -------------------------------------------------------------------
+Mon Jul 18 14:06:06 UTC 2016 - jreidinger@suse.com
+
+- prevent yard failure as it is no longer dependency, so skip
+  ruby autodocs generation if not available ( fix yast-rdp and
```
Comment (lslezak) [inline package/yast2-devtools.changes]: `Style/SpaceInsideParens`: No spaces after ( or before ).

File: build-tools/aminclude/autodocs-ycp.ami (hunk 1 above)
Comment (lslezak) [inline build-tools/aminclude/autodocs-ycp.ami]: Typo: **not** available

File: build-tools/aminclude/autodocs-ycp.ami [hunk 3]
```diff
@@ -47,4 +47,10 @@ index.html: $(AUTODOCS_YCP) $(AUTODOCS_PM) $(AUTODOCS_RB)
 		pod2html --title=$$b --infile=$$i --outfile=$$b.html; \
 		echo "<a href='$$b.html'>$${b/__/::}</a>" > $$b.mod.html; \
 	done
-	if [ -n "$(AUTODOCS_RB)" ]; then $(YARD) doc --title "@RPMNAME@ -- Development Documentation" $(YARD_OPTIONS) $(AUTODOCS_RB); fi
+	if [ -n "$(AUTODOCS_RB)" ]; then \
+          if which yard 2>/dev/null; then \
```
Comment (mvidner) [inline build-tools/aminclude/autodocs-ycp.ami]: s/yard/$(YARD)/ for consistency with the actual run on the next line

**Comment by lslezak** [comment]
LGTM

File: build-tools/aminclude/autodocs-ycp.ami [hunk 4]
```diff
@@ -47,4 +47,10 @@ index.html: $(AUTODOCS_YCP) $(AUTODOCS_PM) $(AUTODOCS_RB)
 		pod2html --title=$$b --infile=$$i --outfile=$$b.html; \
 		echo "<a href='$$b.html'>$${b/__/::}</a>" > $$b.mod.html; \
 	done
-	if [ -n "$(AUTODOCS_RB)" ]; then $(YARD) doc --title "@RPMNAME@ -- Development Documentation" $(YARD_OPTIONS) $(AUTODOCS_RB); fi
+	if [ -n "$(AUTODOCS_RB)" ]; then \
+          if which $(YARD)yard 2>/dev/null; then \
```
Comment (mvidner) [inline build-tools/aminclude/autodocs-ycp.ami]: $(YARD)yard is wrong

**Comment by mvidner** [comment]
LGTM

**Pull request merged**